File: tests/.hooks/typescriptv2/secondary/src/hooks/registration.ts
import { CustomSecurityHook } from "./customsecurityhook.js";
import { Hooks } from "./types.js";

export function initHooks(hooks: Hooks) {
  const customSecurityHook = new CustomSecurityHook();
  hooks.registerBeforeRequestHook(customSecurityHook);
}

File: tests/.hooks/pythonv2/primary/src/openapi/_hooks/registration.py
from .customsecurityhook import CustomSecurityHook
from .types import Hooks
from .testhook import TestHook
from .asynctesthook import AsyncTestSDKInitHook


def init_hooks(hooks: Hooks):
    test_hook = TestHook()
    custom_security_hook = CustomSecurityHook()
    async_init_hook = AsyncTestSDKInitHook()

    hooks.register_sdk_init_hook(test_hook)
    hooks.register_sdk_init_hook(async_init_hook)

    hooks.register_before_request_hook(test_hook)
    hooks.register_before_request_hook(custom_security_hook)

    hooks.register_after_success_hook(test_hook)

    hooks.register_after_error_hook(test_hook)

    hooks.register_after_parse_error_hook(test_hook)
